File: problem-set-1/pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{

    // Prompt user change owed
    float change_owed;
    do
    {
        change_owed = get_float("Change owed: ");
    }
    while (change_owed < 0.009);
    printf("%f\n", change_owed);


    // Rounding change
    float change_rounded = round(change_owed * 100);
    printf("%f\n", change_rounded);


    // Counting coins needed for change
    int coins = 0;

    if (change_rounded > 0)
    {
        // Compairing change to quarter
        int quarter = change_rounded / 25;
        int quarter_val = quarter * 25;
        change_rounded = change_rounded - quarter_val;

        // Compairing change to dime
        int dime = change_rounded / 10;
        int dime_val = dime * 10;
        change_rounded = change_rounded - dime_val;

        // Compairing change to nickel
        int nickel = change_rounded / 5;
        int nickel_val = nickel * 5;
        change_rounded = change_rounded - nickel_val;

        // Compairing change to penny
        int penny = change_rounded / 1;
        int penny_val = penny * 1;
        change_rounded = change_rounded - penny_val;

        // Amount of coins used
        coins = quarter + dime + nickel + penny;

        printf("%i\n", coins);

    }
    else
    {
        printf("%i\n", coins);
    }
}

File: problem-set-6/cash/cash.py
from cs50 import get_float

# Prompt user for change owed
change_owed = get_float("Change owed: ")

# Checking that change is greater than 1 cent
while True:
    if change_owed > 0.009:
        break
    else:
        change_owed = get_float("Change owed: ")

# Rounding change
change_rounded = round(change_owed * 100)

# Counting coins needed for change
coins = 0

# Cointing coins
while change_rounded > 0:

    # Quarter
    q = int(change_rounded / 25)
    q_amount = q * 25
    change_rounded = change_rounded - q_amount
    coins = q
    if change_rounded == 0:
        break

    # Dime
    d = int(change_rounded / 10)
    d_amount = d * 10
    change_rounded = change_rounded - d_amount
    coins = q + d
    if change_rounded == 0:
        break

    # Nickel
    n = int(change_rounded / 5)
    n_amount = n * 5
    change_rounded = change_rounded - n_amount
    coins = q + d + n
    if change_rounded == 0:
        break

    # Penny
    p = int(change_rounded / 1)
    p_amount = p * 1
    change_rounded = change_rounded - p_amount
    coins = q + d + n + p
    if change_rounded == 0:
        break


print(coins)

